server/sched.go

```diff
@@ -654,9 +654,10 @@ func (runner *runnerRef) waitForVRAMRecovery() chan interface{} {
 				totalMemoryNow += gpu.TotalMemory
 				freeMemoryNow += gpu.FreeMemory
 			}
-			// If we're within ~80% of the estimated memory usage recovered, bail out
-			if float32(freeMemoryNow-freeMemoryBefore) > float32(runner.estimatedVRAM)*0.8 {
-				slog.Debug(fmt.Sprintf("gpu VRAM free memory converged after %0.2f seconds", time.Since(start).Seconds()), "model", runner.modelPath)
+			// If we're within ~80% of the estimated memory usage recovered for larger models or ~50% for smaller, bail out
+			if (runner.estimatedVRAM < 2*format.GibiByte && float32(freeMemoryNow-freeMemoryBefore) > float32(runner.estimatedVRAM)*0.5) ||
```

Comment:
A couple of suggestions if I may:

1. Wrap conditional  logic into a helper function with a clear name (e.g., isMemoryRecovered).
2. Use the helper funciton in the main logic block.

```golang
func isMemoryRecovered(runner *Runner, memoryRecovered float32) bool {
    if runner.estimatedVRAM < 2*format.GibiByte {
        return memoryRecovered > float32(runner.estimatedVRAM)*0.5
    }
    return memoryRecovered > float32(runner.estimatedVRAM)*0.8
}

// In the main code:
if isMemoryRecovered(runner, freeMemoryNow-freeMemoryBefore) {
    slog.Debug(fmt.Sprintf("gpu VRAM free memory converged after %0.2f seconds", time.Since(start).Seconds()), 
        "recovered", format.HumanBytes2(freeMemoryNow-freeMemoryBefore), 
        "model", runner.modelPath, 
        "threshold", getThreshold(runner))
    finished <- struct{}{}
    return
}
```

Comment on  the thresholding Logic:

The new threshold for smaller models is set at 50% (0.5), while larger models use the 80% (0.8) threshold. It’s not clear why this division is being made, and without context or data to back this up, it might lead to premature termination for smaller models. This could potentially affect performance for models with memory patterns that don’t conform strictly to this assumption. It would be good to document the reasoning for this decision.